packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+    registry.setAddressFor("Validators", address(validators));
+    validators.setNumRegisteredValidators(1);
+
+    lockedGold.lock.value(value)();
+    registry.setAddressFor("DowntimeSlasher", downtimeSlasher);
+    lockedGold.addSlasher("DowntimeSlasher");
+
+    vm.prank(reporter);
+    accounts.createAccount();
+  }
+
+  function helper_WhenAccountIsSlashedForAllOfItsLockedGold(uint256 penalty, uint256 reward)
+    public
+  {
+    address[] memory lessers = new address[](1);
+    lessers[0] = address(0);
+    address[] memory greaters = new address[](1);
+    greaters[0] = address(0);
+
+    uint256[] memory indices = new uint256[](1);
+    indices[0] = 0;
+
+    vm.prank(downtimeSlasher);
+    lockedGold.slash(address(this), penalty, reporter, reward, lessers, greaters, indices);
+  }
+
+  function test_ShouldReduceAccountsLockedGoldBalance_WhenAccountIsSlashedForAllOfItsLockedGold()
+    public
+  {
+    uint256 penalty = value;
+    uint256 reward = value / 2;
+    helper_WhenAccountIsSlashedForAllOfItsLockedGold(penalty, reward);
+
+    assertEq(lockedGold.getAccountNonvotingLockedGold(address(this)), value - penalty);
+    assertEq(lockedGold.getAccountTotalLockedGold(address(this)), value - penalty);
+  }
+
+  function test_ShouldIncreaseReportersLockedGoldBalance_WhenAccountIsSlashedForAllOfItsLockedGold()
+    public
+  {
+    uint256 penalty = value;
+    uint256 reward = value / 2;
+    helper_WhenAccountIsSlashedForAllOfItsLockedGold(penalty, reward);
+
+    assertEq(lockedGold.getAccountNonvotingLockedGold(reporter), reward);
+    assertEq(lockedGold.getAccountTotalLockedGold(reporter), reward);
+  }
+
+  function test_ShouldIncreaseCommunityFundLockedGoldBalance_WhenAccountIsSlashedForAllOfItsLockedGold()
+    public
+  {
+    uint256 penalty = value;
+    uint256 reward = value / 2;
+    helper_WhenAccountIsSlashedForAllOfItsLockedGold(penalty, reward);
+
+    assertEq(address(governance).balance, penalty - reward);
+  }
+
+  // original tests regarding delegating are already covered in other tests below
+
+  function test_ShouldRevert_WhenTheSlashingContractIsRemovedFromIsSlasher() public {
+    uint256 penalty = value;
+    uint256 reward = value / 2;
+    lockedGold.removeSlasher("DowntimeSlasher", 0);
+    vm.expectRevert("Caller is not a whitelisted slasher.");
+    helper_WhenAccountIsSlashedForAllOfItsLockedGold(penalty, reward);
+  }
+
+  function test_ShouldReduceAccountsNonVotingLockedGoldBalance_WhenAccountIsSlashedForOnlyItsNonvotingBalance_WhenTheAccountHasHalfVotingAndHalfNonVotingGold()
+    public
+  {
+    uint256 voting = value / 2;
+    uint256 nonVoting = value - voting;
+    uint256 penalty = nonVoting;
+    uint256 reward = penalty / 2;
+    electionSlashTest.vote(group, voting, address(0), address(0));
+
+    helper_WhenAccountIsSlashedForAllOfItsLockedGold(penalty, reward);
+    assertEq(lockedGold.getAccountNonvotingLockedGold(address(this)), nonVoting - penalty);
```

Comment:
Shouldn't we also check the voting gold?

Reply:
We dont have method to check voting gold - it is necessary to get totalLockedGold and subtract nonVotingGold  (please note that this is Election voting rather then Governance voting)